src/components/common/ErrorBoundary.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+class ErrorBoundary extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { error: null, errorInfo: null };
+    }
+
+    componentDidCatch(error, errorInfo) {
+        // Catch errors in any components below and re-render with error message
+        this.setState({
+            error: error,
+            errorInfo: errorInfo
+        })
+        // You can also log error messages to an error reporting service here
+    }
+
+    render() {
+        if (this.state.errorInfo) {
+            // Error path
```

Comment:
I believe comments needed when they add some clearance to code, but in this case it's pretty clear what is going on

Reply:
Fixed, thank you.